src/app/Mappings/components/MappingBuilder/MappingBuilder.tsx

```diff
@@ -58,7 +58,7 @@ export const MappingBuilder: React.FunctionComponent<IMappingBuilderProps> = ({
         <Text component="p">{instructionText}</Text>
       </TextContent>
       {builderItems.map((item, itemIndex) => {
-        const key = item.source ? `${item.source.id}` : `empty-${itemIndex}`;
+        const key = item.source ? `${item.source.id}-${itemIndex}` : `empty-${itemIndex}`;
```

Comment:
We should also update our `MOCK_VMWARE_DATASTORES` data so the ids are actually different in each one, which would make this unnecessary-- technically if the items in this list change order or one is removed etc the key is supposed to remain the same for the same object it represents, so React doesn't have to re-render its children. But these components are trivial to render so it's fine